Fix: Fixed align right issue on avatar component

Mayank-Tripathi32 opened the pull request:
resolves #67493

## What?
Added styles for Align Right in editor.scss for block themes to resolve the issue where the Avatar component alignment was not working when the "Link to Profile" option was enabled.

## Why?
The Align Right feature for the Avatar component was not functioning as expected in block-based themes (e.g., Twenty Twenty-Four, Twenty Twenty-Five) when the "Link to Profile" option was selected. This PR addresses that issue by ensuring that the alignment works correctly across block themes, ensuring consistency with classic themes like Twenty Twenty-One, where it worked fine.

## How?
This PR modifies the editor.scss file to explicitly add and enforce styles for the Align Right option in the block editor for Avatar components.

It adds following style snippet.

```css
.wp-block-avatar.alignright {
	.components-resizable-box__container {
		float: right;
	}
}
```

## Testing Instructions

1. Activate a block-based theme, e.g., Twenty Twenty-Four or Twenty Twenty-Five.
2. Add the Avatar component to a post or page using the block editor.
3. Enable the "Link to Profile" option in the Avatar component settings.
4. Set the alignment option to "Align Right".
5. Save and preview the post/page.
6. Repeat for classic theme.

## Screencast

### Before
https://github.com/user-attachments/assets/bab1c8e3-f2f8-43d2-a484-4481569e6cea

### After

https://github.com/user-attachments/assets/c2f35d91-25ae-4f15-b653-1380831a77bd

Mayank-Tripathi32 committed:
feat: fix right align issue in block themes

ellatrix reviewed (dismissed):
Hi! Quickly looking at the output of the Avatar block, it seems that the whole block is wrapped in the link, and the link is not part of the block itself. I think fixing that would be the correct approach. Would you be able to try that?

Mayank-Tripathi32 commented:
> Hi! Quickly looking at the output of the Avatar block, it seems that the whole block is wrapped in the link, and the link is not part of the block itself. I think fixing that would be the correct approach. Would you be able to try that?

Yes, I will work on it today.

Mayank-Tripathi32 committed:
feat: fixed edit.js link wrapper

Mayank-Tripathi32 committed:
Revert "feat: fix right align issue in block themes"

This reverts commit edab7a5ec7ba01b018e91f2df6206d2926f3e6dd.

Mayank-Tripathi32 commented:
Hello @ellatrix,
Thank you for your feedback. I’ve addressed the issue you highlighted regarding the wrapper in the editor incorrectly wrapping the block. I’ve implemented the necessary adjustments and would appreciate it if you could review the changes.

I have also reverted the initial commit related to css changes.

youknowriad reviewed (approved):
This makes a lot of sense.
